src/func.d

```diff
@@ -1791,13 +1791,14 @@ public:
                         fbody = new CompoundStatement(Loc(), fbody, s);
                     }
                 }
-                if (returns && !fbody.isErrorStatement())
+                if (returns)
```

Comment:
Sorry for late reply. I think, we should have same semantic analysis behavior with/without gagging. Branching analysis process depending on `global.gag` would be a cause of bug.